### PR TITLE
[crypto] add BLS12381 verify native function in move

### PR DESCRIPTION
**Pull request by punwai**

### CODE REVIEW

File: crates/sui-framework/src/natives/crypto.rs [hunk 1]
```diff
@@ -40,3 +40,38 @@ pub fn ecrecover(
...
+    };
+
+    let public_key = match <narwhal_crypto::bls12381::BLS12381PublicKey as ToFromBytes>::from_bytes(
+        &public_key_bytes,
+    ) {
+        Ok(public_key) => public_key,
+        Err(_) => return Ok(NativeResult::ok(cost, smallvec![Value::bool(false)])),
+    };
+
+    match public_key.verify(&msg, &signature) {
```
Comment (joyqvq) [inline crates/sui-framework/src/natives/crypto.rs:73]: is this the raw message or the hashed message? maybe worth adding in the comment

**Comment by joyqvq** [comment]
lgtm after rebase and a small comment on expected message form.

File: crates/sui-framework/src/natives/crypto.rs (hunk 1 above)
Comment (punwai) [reply crates/sui-framework/src/natives/crypto.rs:73]: It is the raw message - does it have to be a hash?

Comment (punwai) [reply crates/sui-framework/src/natives/crypto.rs:73]: That is also fine to do a hash - I only need the hash anyways.

File: crates/sui-framework/src/natives/crypto.rs [hunk 2]
```diff
@@ -40,3 +40,38 @@ pub fn ecrecover(
         Err(_) => Ok(NativeResult::err(cost, INVALID_SIGNATURE)),
     }
 }
+
+pub fn bls12381_verify(
```
Comment (kchalkias) [inline crates/sui-framework/src/natives/crypto.rs:44]: We should add some docs here mainly explaining that this only works for G1 sigs, G2 keys

**Review by kchalkias — APPROVED** [review]
Please see my comment re explaining that this is BLS where the key is in G2 and sig in G1 (probably change the name of the function to make this clear to). These can happen in an upcoming PR

File: crates/sui-framework/src/natives/crypto.rs [hunk 3]
```diff
@@ -40,3 +40,38 @@ pub fn ecrecover(
...
+    context: &mut NativeContext,
```
Comment (oxade) [inline crates/sui-framework/src/natives/crypto.rs:45]: @punwai please make an issue for calibrating the cost of this native and assign to me

File: crates/sui-framework/src/natives/crypto.rs (hunk 2 above)
Comment (punwai) [reply crates/sui-framework/src/natives/crypto.rs:44]: Yep, for sure - I'll add a doc here mentioning this, but also refer them to crypto.move, which also says this but in greater detail.

File: crates/sui-framework/src/natives/crypto.rs (hunk 3 above)
Comment (punwai) [reply crates/sui-framework/src/natives/crypto.rs:45]: Will do, thanks¡

**Pull request merged**